From 571f51c424ea50c31b85003fa68f420f0573065f Mon Sep 17 00:00:00 2001
From: Noel Grandin <noel@peralex.com>
Date: Mon, 7 Dec 2015 11:10:03 +0200
Subject: [PATCH] fix powerpc build

fallout from the wide char changes

<sberg> noelgrandin, _rene_, ah, yeah, that's something that broke with 0b07406f7147b9abbb2095d9e038b13293cb8b10
<IZBot> core - Use C++11 char16_t for sal_Unicode - http://cgit.freedesktop.org/libreoffice/core/commit/?id=0b07406f7147b9abbb2095d9e038b13293cb8b10

Change-Id: If4303bd1d8577612250b1857b809b022d13759e4
(cherry picked from commit b2d145193e34b57ca991063fc7c3cf8200339dea)
---
 include/oox/helper/helper.hxx | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/include/oox/helper/helper.hxx b/include/oox/helper/helper.hxx
index 516c7a2..5029d5f 100644
--- a/include/oox/helper/helper.hxx
+++ b/include/oox/helper/helper.hxx
@@ -218,6 +218,9 @@ public:
 #ifdef OSL_BIGENDIAN
     static void  convertLittleEndian( sal_Int8& ) {}     // present for usage in templates
     static void  convertLittleEndian( sal_uInt8& ) {}    // present for usage in templates
+#if !defined SAL_W32 || defined __MINGW32__ // cf. sal/types.h sal_Unicode
+    static void  convertLittleEndian( char16_t& rnValue )   { swap2( reinterpret_cast< sal_uInt8* >( &rnValue ) ); }
+#endif
     static void  convertLittleEndian( sal_Int16& rnValue )  { swap2( reinterpret_cast< sal_uInt8* >( &rnValue ) ); }
     static void  convertLittleEndian( sal_uInt16& rnValue ) { swap2( reinterpret_cast< sal_uInt8* >( &rnValue ) ); }
     static void  convertLittleEndian( sal_Int32& rnValue )  { swap4( reinterpret_cast< sal_uInt8* >( &rnValue ) ); }
-- 
2.5.0

